src/analyzer.ts

```diff
@@ -432,35 +568,100 @@ export class Analyzer {
       declaringFile: packageJson,
     };
     Object.assign(placeholder, remainingConfig);
+    return {ok: true, value: undefined};
   }
 
   #checkForCyclesAndSortDependencies(
     config: ScriptConfig,
     trail: Set<ScriptReferenceString>
-  ) {
+  ): Result<void> {
     const trailKey = scriptReferenceToString(config);
+    const supplementalLocations: MessageLocation[] = [];
     if (trail.has(trailKey)) {
       // Found a cycle.
-      const trailArray = [];
       let cycleStart = 0;
-      // Trail is in graph traversal order because JavaScript Set iteration
+      // Trail is in graph traversal order because JavaScript Map iteration
       // order matches insertion order.
       let i = 0;
-      for (const visited of trail) {
-        trailArray.push(stringToScriptReference(visited));
-        if (visited === trailKey) {
+      for (const visitedKey of trail) {
+        if (visitedKey === trailKey) {
           cycleStart = i;
         }
         i++;
       }
-      trailArray.push({packageDir: config.packageDir, name: config.name});
-      throw new WireitError({
-        type: 'failure',
-        reason: 'cycle',
-        script: config,
-        length: trail.size - cycleStart,
-        trail: trailArray,
+      const trailArray = [...trail].map((key) => {
+        const placeholder = this.#placeholders.get(key);
+        if (placeholder == null) {
+          throw new Error(
+            `Internal error: placeholder not found for ${key} during cycle detection`
+          );
+        }
+        return placeholder as ScriptConfig;
       });
+      trailArray.push(config);
+      const cycleEnd = trailArray.length - 1;
```

Comment:
Needs a rebase? Seeing changes from the "Convert other errors to diagnostics when possible" PR here.

Reply:
Yep, rebased